src/lib.rs

```diff
@@ -190,35 +305,39 @@ pub fn run(args: Args) {
                             })
                             .collect::<Vec<_>>();
                         match successfully_parsed_query_languages.len() {
-                            0 => return,
+                            0 => {
+                                return NonFatalSearchError::QueryNotParseableForFile {
+                                    path: project_file_dir_entry.path().to_owned(),
+                                }
+                                .into();
+                            }
                             1 => successfully_parsed_query_languages[0],
                             _ => {
-                                error_disambiguate_language_for_file(
-                                    &project_file_dir_entry,
-                                    &successfully_parsed_query_languages,
-                                );
-                                return;
+                                return NonFatalSearchError::AmbiguousLanguageForFile {
+                                    path: project_file_dir_entry.path().to_owned(),
+                                    languages: successfully_parsed_query_languages,
+                                }
+                                .into();
                             }
                         }
                     }
                 },
             };
-            let query = return_if_none!(
-                cached_queries.get_and_cache_query_for_language(&query_text, language)
-            );
-            let capture_index = capture_index.get_or_init(&query, args.capture_name.as_deref());
+            let query = cached_queries
+                .get_and_cache_query_for_language(&query_text, language)
+                .ok_or_else(|| NonFatalSearchError::QueryNotParseableForFile {
+                    path: project_file_dir_entry.path().to_owned(),
+                })?;
+            let capture_index = capture_index
+                .get_or_init(&query, args.capture_name.as_deref())
+                .map_err(Error::from)?;
```

Comment:
Couldn't it implement that too?

Reply:
Sure why not, updated to implement using `From` to go "directly" from a `CaptureIndexError` to a `SingleFileSearchError`

It seems like it would be nice if there was some "baked-in concept" for these types of "multiple levels of automated mappings" but I don't know what that would look like

So in the meantime I guess ya why not just explicitly implement the [whatever the math word is for if you have `a -> b` and `b -> c` then you have `a -> c`]